syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionIrcConnectorJSONLoadingTest.java

```diff
@@ -0,0 +1,180 @@
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.fusesource.ide.syndesis.extensions.core.model;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import org.fusesource.ide.syndesis.extensions.core.model.ActionDataShape;
+import org.fusesource.ide.syndesis.extensions.core.model.PropertyDefinitionStep;
+import org.fusesource.ide.syndesis.extensions.core.model.SyndesisAction;
+import org.fusesource.ide.syndesis.extensions.core.model.SyndesisActionDescriptor;
+import org.fusesource.ide.syndesis.extensions.core.model.SyndesisExtension;
+import org.fusesource.ide.syndesis.extensions.core.model.SyndesisExtensionProperty;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author lhein
+ */
+public class SyndesisExtensionIrcConnectorJSONLoadingTest {
+
+	private SyndesisExtension extension;
+
+	@Before
+	public void onStart() throws IOException {
+		File jsonFile = new File("testdata/irc-connector.json");
+		this.extension = SyndesisExtension.getJSONFactoryInstance(new FileInputStream(jsonFile));
+	}
+
+	/**
+	 * Test method for {@link org.fusesource.ide.syndesis.extensions.core.model.SyndesisExtension#getExtensionId()}.
+	 */
+	@Test
+	public void testGetExtensionId() {
+		assertEquals("extensionId value is not matching", "io.syndesis.extensions:syndesis-connector-irc", extension.getExtensionId());
+	}
+
+	/**
+	 * Test method for {@link org.fusesource.ide.syndesis.extensions.core.model.SyndesisExtension#getVersion()}.
+	 */
+	@Test
+	public void testGetVersion() {
+		assertEquals("version value is not matching", "1.0.0", extension.getVersion());
+	}
+
+	/**
+	 * Test method for {@link org.fusesource.ide.syndesis.extensions.core.model.SyndesisExtension#getName()}.
+	 */
+	@Test
+	public void testGetName() {
+		assertEquals("name value is not matching", "IRC", extension.getName());
+	}
+
+	/**
+	 * Test method for {@link org.fusesource.ide.syndesis.extensions.core.model.SyndesisExtension#getDescription()}.
+	 */
+	@Test
+	public void testGetDescription() {
+		assertEquals("description value is not matching", "IRC Extension Connector for Syndesis", extension.getDescription());
+	}
+	
+	/**
+	 * Test method for {@link org.fusesource.ide.syndesis.extensions.core.model.SyndesisExtension#getDependencies()}.
+	 */
+	@Test
+	public void testGetDependencies() {
+		assertThat(extension.getDependencies()).filteredOn("id", "io.syndesis.integration:integration-runtime:jar:1.3-SNAPSHOT").isNotNull(); 
```

Comment:
to write it a little simpler, a first solution would be:
```
	@Test
	public void testGetDependencies() {
		dependenciesContains("io.syndesis.integration:integration-runtime:jar:1.3-SNAPSHOT");
		dependenciesContains("org.apache.camel:camel-core:jar:2.20.1");
		dependenciesContains("org.apache.camel:camel-irc:jar:2.20.1");
		dependenciesContains("org.apache.camel:camel-spring-boot:jar:2.20.1");
		dependenciesContains("org.springframework.boot:spring-boot-autoconfigure:jar:1.5.8.RELEASE");
		dependenciesContains("org.springframework.boot:spring-boot-loader:jar:1.5.8.RELEASE");
	}

	private void dependenciesContains(String id) {
		assertThat(extension.getDependencies()).filteredOn("id", id).isNotNull();
	}
```

Reply:
a second possibility is:
```
		List<String> idOfDependencies = extension.getDependencies().stream().map(edep -> edep.getId()).collect(Collectors.toList());
		assertThat(idOfDependencies)
		.contains(
				"io.syndesis.integration:integration-runtime:jar:1.3-SNAPSHOT",
				"org.apache.camel:camel-core:jar:2.20.1",
				"org.apache.camel:camel-irc:jar:2.20.1",
				"org.apache.camel:camel-spring-boot:jar:2.20.1",
				"org.springframework.boot:spring-boot-autoconfigure:jar:1.5.8.RELEASE",
				"org.springframework.boot:spring-boot-loader:jar:1.5.8.RELEASE");
```
I think this one is better because it will provide a better failure message with all non-matching elements in a single time

Reply:
nice